Mathlib/CategoryTheory/ChosenFiniteProducts/Cat.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Nicolas Rolland. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolas Rolland
+-/
+import Mathlib.CategoryTheory.ChosenFiniteProducts
+/-!
+# Chosen finite products in `Cat`
+
+This file proves that the cartesian product of a pair of categories agrees with the
+product in `Cat`, and provides the associated `ChosenFiniteProducts` instance.
+-/
+
+universe v u
+
+namespace CategoryTheory
+
+namespace Cat
+
+open Limits
+
+/-- The chosen terminal object in `Cat`. -/
+abbrev chosenTerminal : Cat := Cat.of (ULift (ULiftHom (Discrete Unit)))
+
+example : chosenTerminal := ⟨⟨⟨⟩⟩⟩
+
+/-- The chosen terminal object in `Cat` is terminal. -/
+def chosenTerminalIsTerminal : IsTerminal chosenTerminal :=
+  IsTerminal.ofUniqueHom (fun _ ↦ (Functor.const _).obj ⟨⟨⟨⟩⟩⟩) fun _ _ ↦ rfl
+
+/-- The chosen product of categories `C × D` yields a product cone in `Cat`. -/
+def prodCone (C D : Cat.{v,u}) : BinaryFan C D :=
+  .mk (P := .of (C × D)) (Prod.fst _ _) (Prod.snd _ _)
+
+/-- The product cone in `Cat` is indeed a product. -/
+def isLimitProdCone (X Y : Cat) : IsLimit (prodCone X Y) := BinaryFan.isLimitMk
+  (fun S => S.fst.prod' S.snd) (fun _ => rfl) (fun _ => rfl) (fun _ _ h1 h2 =>
+    Functor.hext
+      (fun _ ↦ Prod.ext (by simp [← h1]) (by simp [← h2]))
+      (fun _ _ _ ↦ by dsimp; rw [← h1, ← h2]; rfl))
+
+instance : ChosenFiniteProducts Cat where
+  product (X Y : Cat) := { isLimit := isLimitProdCone X Y }
+  terminal  := { isLimit := chosenTerminalIsTerminal }
+
+example : MonoidalCategory Cat := by infer_instance
+
+example : SymmetricCategory Cat := by infer_instance
+
+end Cat
+
+namespace Monoidal
+
+open MonoidalCategory
+
+lemma tensorObj (C : Cat) (D : Cat) : C ⊗ D = Cat.of (C × D) := rfl
+
+lemma whiskerLeft (X : Cat) {A : Cat} {B : Cat} (f : A ⟶ B) :
+    X ◁ f = (𝟭 X).prod f := rfl
+
+lemma whiskerLeft_fst (X : Cat) {A : Cat} {B : Cat} (f : A ⟶ B) :
+    (X ◁ f) ⋙ Prod.fst _ _ = Prod.fst _ _ := rfl
+
+lemma whiskerLeft_snd (X : Cat) {A : Cat} {B : Cat} (f : A ⟶ B) :
+    (X ◁ f) ⋙ Prod.snd _ _ = Prod.snd _ _ ⋙ f := rfl
+
+lemma whiskerRight {A : Cat} {B : Cat} (f : A ⟶ B)  (X : Cat) :
+    f ▷  X  = f.prod (𝟭 X) := rfl
+
+lemma whiskerRight_fst {A : Cat} {B : Cat} (f : A ⟶ B)  (X : Cat) :
+    (f ▷ X) ⋙ Prod.fst _ _  = Prod.fst _ _ ⋙ f := rfl
+
+lemma whiskerRight_snd {A : Cat} {B : Cat} (f : A ⟶ B)  (X : Cat) :
+    (f ▷ X) ⋙ Prod.snd _ _  = Prod.snd _ _ := rfl
+
+lemma tensorHom {A : Cat} {B : Cat} (f : A ⟶ B)  {X : Cat} {Y : Cat} (g : X ⟶ Y) :
```

Comment:
Could you fix the whitespace? (There are many other instances to fix.)
```suggestion
lemma tensorHom {A : Cat} {B : Cat} (f : A ⟶ B) {X : Cat} {Y : Cat} (g : X ⟶ Y) :
```

Reply:
I think I fixed them all. is there a linter somewhere for this ?